packages/client/src/client.ts

```diff
@@ -1,48 +1,61 @@
-import { EvaluationContext, OpenFeatureError } from '@openfeature/shared';
-import { SafeLogger } from '@openfeature/shared';
 import {
   ClientMetadata,
   ErrorCode,
+  EvaluationContext,
   EvaluationDetails,
   FlagValue,
   FlagValueType,
   HookContext,
   JsonValue,
   Logger,
+  OpenFeatureError,
+  ProviderStatus,
   ResolutionDetails,
-  StandardResolutionReasons
+  SafeLogger,
+  StandardResolutionReasons,
 } from '@openfeature/shared';
 import { OpenFeature } from './open-feature';
-import { Client, FlagEvaluationOptions, Handler, Hook, OpenFeatureEventEmitter, Provider, ProviderEvents } from './types';
+import {
+  Client,
+  FlagEvaluationOptions,
+  Handler,
+  Hook,
+  OpenFeatureEventEmitter,
+  Provider,
+  ProviderEvents,
+} from './types';
 
 type OpenFeatureClientOptions = {
   name?: string;
   version?: string;
 };
 
 export class OpenFeatureClient implements Client {
-  readonly metadata: ClientMetadata;
   private _hooks: Hook[] = [];
   private _clientLogger?: Logger;
 
   constructor(
     // functions are passed here to make sure that these values are always up to date,
-    // and so we don't have to make these public properties on the API class. 
+    // and so we don't have to make these public properties on the API class.
     private readonly providerAccessor: () => Provider,
-    private readonly providerReady: () => boolean,
     private readonly events: () => OpenFeatureEventEmitter,
     private readonly globalLogger: () => Logger,
-    options: OpenFeatureClientOptions,
-  ) {
-    this.metadata = {
-      name: options.name,
-      version: options.version,
-    } as const;
+    private readonly options: OpenFeatureClientOptions
+  ) {}
+
+  get metadata(): ClientMetadata {
+    return {
+      name: this.options.name,
+      version: this.options.version,
+      provider: this.providerAccessor().metadata,
+    };
   }
 
   addHandler(eventType: ProviderEvents, handler: Handler): void {
     this.events().on(eventType, handler);
-    if (eventType === ProviderEvents.Ready && this.providerReady()) {
+    const providerReady = !this._provider.status || this._provider.status === ProviderStatus.READY;
```

Comment:
I'm assuming `!this._provider.status` is there to support backward compatibility for providers that haven't set a status. If that's the case, could you add a comment stating that? This is something that may be confusing to future devs if they don't understand the history.

Reply:
Yeah exactly.
This is from Requirement 2.4.2:  `Providers without this field can be assumed to be ready immediately.`
I surely can add a comment.

Reply:
I understand why you did it. I agree lots of comment doesn't hurt.

Reply:
So should we do something here?